src/@next/Select/components/Activator/ActivatorSelect.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useRef } from 'react';
+import { Icon } from '../../../Icon';
+
+import { Typography } from '../../../Typography';
+import { Blue, Neutral } from '../../../utilities/colors';
+import {
+  ActivatorSelectContextProps,
+  useSelectActivator,
+} from './ActivatorContext';
+import { StyledSelect, StyledTag, TagsContainer } from './ActivatorStyle';
+
+export interface ActivatorSelectProps
+  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  value?: string;
+  placeholder?: string;
+  onRemoveTag?({ option }: { option: string }): void;
+  width?: number;
+}
+export const ActivatorSelect = ({
+  placeholder,
+  value,
+  onClick,
+  onRemoveTag,
+  width,
+  ...props
+}: ActivatorSelectProps) => {
+  const activatorRef = useRef(null);
+  const hasValue = !!value;
+  const activatorContext = useSelectActivator() as ActivatorSelectContextProps;
```

Comment:
I'm not sure if we can do either? since we are using 2 different interfaces. 
I think this is the reason why typescript is unable to identify which one I'm currently using: https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#unions-with-common-fields

Reply:
Hmm probably something like this? https://ordina-jworks.github.io/architecture/2021/02/12/react-generic-context.html

But upon looking at the ActivatorProvider type I'm more concern of how different type they are but living in the same provider, I'm leaning forward to actually separate them into different providers altogether

Reply:
thanks for the insights, Putra. I agree with separating them. I've addressed the separation